pkg/codegen/extension.go

```diff
@@ -6,59 +6,55 @@ import (
 )
 
 const (
-	extPropGoType    = "x-go-type"
-	extPropOmitEmpty = "x-omitempty"
-	extPropExtraTags = "x-go-extra-tags"
-	extMiddlewares   = "x-go-middlewares"
+	extPropGoType        = "x-go-type"
+	extPropOmitEmpty     = "x-omitempty"
+	extPropExtraTags     = "x-go-extra-tags"
+	extPropOptionalValue = "x-go-optional-value"
+	extPropString        = "x-go-string"
+	extMiddlewares       = "x-go-middlewares"
 )
 
 func extTypeName(extPropValue interface{}) (string, error) {
-	raw, ok := extPropValue.(json.RawMessage)
-	if !ok {
-		return "", fmt.Errorf("failed to convert type: %T", extPropValue)
-	}
 	var name string
-	if err := json.Unmarshal(raw, &name); err != nil {
-		return "", fmt.Errorf("failed to unmarshal json: %w", err)
-	}
-
-	return name, nil
+	err := extParseAny(extPropValue, &name)
+	return name, err
 }
 
 func extParseOmitEmpty(extPropValue interface{}) (bool, error) {
-	raw, ok := extPropValue.(json.RawMessage)
-	if !ok {
-		return false, fmt.Errorf("failed to convert type: %T", extPropValue)
-	}
-
-	var omitEmpty bool
-	if err := json.Unmarshal(raw, &omitEmpty); err != nil {
-		return false, fmt.Errorf("failed to unmarshal json: %w", err)
-	}
-
-	return omitEmpty, nil
+	return extParseBool(extPropValue)
 }
 
 func extExtraTags(extPropValue interface{}) (map[string]string, error) {
-	raw, ok := extPropValue.(json.RawMessage)
-	if !ok {
-		return nil, fmt.Errorf("failed to convert type: %T", extPropValue)
-	}
 	var tags map[string]string
-	if err := json.Unmarshal(raw, &tags); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal json: %w", err)
-	}
-	return tags, nil
+	err := extParseAny(extPropValue, &tags)
+	return tags, err
 }
 
 func extParseMiddlewares(extPropValue interface{}) ([]string, error) {
+	var middlewares []string
+	err := extParseAny(extPropValue, &middlewares)
+	return middlewares, err
+}
+
+func extParseOptionalValue(extPropValue interface{}) (bool, error) {
```

Comment:
This function is unused. I'm not sure if it was before, but if generation passes we might as well remove it.

Reply:
I've removed both `extParseOptionalValue` and `extParseOmitEmpty` while I'm at it, since `extParseBool` effectively does that.